indexer/ai/src/main/java/com/google/cloudsearch/ai/AISkill.java

```diff
@@ -0,0 +1,39 @@
+package com.google.cloudsearch.ai;
+
+
+import java.util.List;
+
+import com.google.common.collect.Multimap;
+import org.json.simple.JSONObject;
+
+/**
+    Interface for all Skills.
+ */
+public interface AISkill{
+
+    public void setAISkillName(String aiSkillName);
+    public void setOutputMapping(List<Mapping> outputMapping);
+    public void setInputs(JSONObject input);
+    public void setFilter(JSONObject filter);
+    public String getAISkillName();
+    public List<Mapping> getOutputMapping();
+    public JSONObject getInputs();
+    public JSONObject getFilter();
+
+    /**
+     * Parse the JSONObject for the skill in the Configuration.
+     * Call the setter functions and initialize all the data required for the skill.
+     * TODO: Decide on the type of error or exception if the config is invalid
+     *
+     * @param aiSkill        JSONObject for the skill
+     */
+    public void parse(JSONObject aiSkill);
```

Comment:
This should be protected method, we don't need to invoke parse from outside. Can be invoked from the constructor.